dbt/include/trino/macros/materializations/table.sql

```diff
@@ -48,20 +48,28 @@
 {% macro on_table_exists_logic(on_table_exists, existing_relation, intermediate_relation, backup_relation, target_relation) -%}
   {#-- Create table with given `on_table_exists` mode #}
   {% if on_table_exists == 'rename' %}
-      {#-- build modeldock #}
-      {% call statement('main') -%}
-        {{ create_table_as(False, intermediate_relation, sql) }}
-      {%- endcall %}
 
-      {#-- cleanup #}
-      {% if existing_relation is not none %}
-          {{ adapter.rename_relation(existing_relation, backup_relation) }}
+      {#-- table does not exists #}
+      {% if existing_relation is none %}
+          {% call statement('main') -%}
+              {{ create_table_as(False, target_relation, sql) }}
+          {%- endcall %}
       {% endif %}
 
-      {{ adapter.rename_relation(intermediate_relation, target_relation) }}
+      {#-- table does exists #}
+      {% if existing_relation is not none %}
```

Comment:
```suggestion
      {% else %}
```

change to `else`, remove `endif` above